ta4j-core/src/main/java/org/ta4j/core/util/BarAggregator.java

```diff
@@ -0,0 +1,110 @@
+package org.ta4j.core.util;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.ta4j.core.Bar;
+import org.ta4j.core.BaseBar;
+import org.ta4j.core.BaseTimeSeries;
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.num.Num;
+
+public class BarAggregator {
+	
+	private BarAggregator() {}
+
+	/**
+	 * Aggregates the bars of the <code>barSeries</code> by
+	 * <code>timePeriod</code>. The new <code>timePeriod</code> must be a
+	 * multiple of the actual time period.
+	 * 
+	 * @param barSeries the actual TimeSeries
+	 * @param timePeriod the actual timePeriod
+	 * @return the aggregated TimeSeries
+	 */
+	public static TimeSeries aggregateBarSeries(TimeSeries barSeries, Duration timePeriod) {
+		String name = barSeries.getName() + "_upscaled_to_" + timePeriod;
+		return aggregateBarSeries(name, barSeries, timePeriod);
+	}
+
+	/**
+	 * Aggregates the bars of the <code>barSeries</code> by
+	 * <code>timePeriod</code>. The new <code>timePeriod</code> must be a
+	 * multiple of the actual time period.
+	 * 
+	 * @param name       the name of the returned TimeSeries
+	 * @param barSeries the actual TimeSeries
+	 * @param timePeriod the actual timePeriod
+	 * @return the aggregated TimeSeries
+	 */
+	public static TimeSeries aggregateBarSeries(String name, TimeSeries barSeries, Duration timePeriod) {
+		List<Bar> sumBars = aggregateBars(barSeries.getBarData(), timePeriod);
+		return new BaseTimeSeries(name, sumBars);
+	}
+
+	/**
+	 * Aggregates a list of bars by <code>timePeriod</code>.The new
+	 * <code>timePeriod</code> must be a multiple of the actual time period.
+	 * 
+	 * @param bars       the actual bars
+	 * @param timePeriod the new timePeriod
+	 * @return the aggregated bars with new <code>timePeriod</code>
+	 */
+	public static List<Bar> aggregateBars(List<Bar> bars, Duration timePeriod) {
+		List<Bar> sumBars = new ArrayList<>();
+		if (bars.isEmpty()) return sumBars;
```

Comment:
Please use the same convention with bracket for statements. In one you are using brackets, in the other one no:
instead of:
`if (bars.isEmpty()) return sumBars`


Should be:
`if (bars.isEmpty()) {

    return sumBars

}`